### PR TITLE
Introduce Preliminary Guideline Document for Managing Java Key Stores and Certificate Trust Stores

### LINKED ISSUE #14: Introduce Guideline Document for Managing Java Key Stores and Certificate Trust Stores
**Description:**
It is required to introduce a guideline documentation which discusses about managing internal Java key stores and trust stores in container based WSO2 product deployments.

Suggested topics are as follows:
- [x] How to add a customized keystore to a container based deployment?
- [x] How to add a customized truststore to a container based deployment?
- [x] Approach in Kubernetes (note)
- [x] Peer review

**Related Issues:**
https://github.com/wso2/kubernetes-apim/issues/367, https://github.com/wso2/kubernetes-apim/issues/378, https://github.com/wso2/kubernetes-apim/issues/401, https://github.com/wso2/kubernetes-is/issues/224

### PR DESCRIPTION
**Pull request by chirangaalwis**

### CODE REVIEW

File: deploy/Managing_Keystores_And_Truststores.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+Please refer the official WSO2 [recommendations](https://docs.wso2.com/display/ADMIN44x/Using+Asymmetric+Encryption#UsingAsymmetricEncryption-recommendationsRecommendationsforsettingupkeystoresinWSO2products)
+for managing keystores prior to following this guide.
+
+### Manage Keystores and Certificate Truststores in Container Based Deployments
+
+As per aforementioned WSO2 [recommendations](https://docs.wso2.com/display/ADMIN44x/Using+Asymmetric+Encryption#UsingAsymmetricEncryption-recommendationsRecommendationsforsettingupkeystoresinWSO2products),
+it is often required to customize the default, WSO2 product keystore and truststore in a production grade deployment.
+
+The most appropriate approach for introducing a custom keystore or a truststore to a container based deployment is via
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]: is by using

File: deploy/Managing_Keystores_And_Truststores.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+volume mounting.
+
+1. Prepare the custom Java keystore.
+
+   Please follow the official guidelines for [creating a new keystore](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-Creatinganewkeystore)
+   or [adding a Certificate Authority signed certificate to an existing keystore](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-ca_certificateAddingCA-signedcertificatestokeystores)
+   as per your preference.
+   
+   Also, please follow the official [documentation](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-Step3:Importingcertificatestothetruststore)
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]: To import certificates to a truststore follow this [documentation](https://docs.wso2.com/display/ADMIN44x/Creating+New+Keystores#CreatingNewKeystores-Step3:Importingcertificatestothetruststore)

File: deploy/Managing_Keystores_And_Truststores.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+   for guidelines required to import a certificate to a truststore.
+   
+2. Add the keystore to the product container via a volume mount depending on the container platform.
+
+   We recommend you to use the volume mount point allocated for non-configuration resources in WSO2 product containers
+   (i.e. `/home/wso2carbon/wso2-artifact-volume`) for this purpose.
+   
+   In Carbon Kernel version 4 based products, the default `wso2carbon.jks` keystore is located at the
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]: give a highlight to **version 4** for this point and next to help skim and scan

File: deploy/Managing_Keystores_And_Truststores.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+   `<WSO2_PRODUCT_HOME>/repository/resources/security` directory. Thus, you may mount the volume containing the
+   custom keystore to `/home/wso2carbon/wso2-artifact-volume/repository/resources/security` directory.
+   
+   In Carbon Kernel version 5 based products, the default `wso2carbon.jks` keystore is located at the
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]: **version 5**

File: deploy/Managing_Keystores_And_Truststores.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+   `<WSO2_PRODUCT_HOME>/resources/security` directory. Thus, you may mount the volume containing the custom keystore
+   to `/home/wso2carbon/wso2-artifact-volume/resources/security` directory.
+   
+3. Add the truststore to the product container via a volume mount depending on the container platform.
+
+   We recommend you to use the volume mount point allocated for non-configuration resources in WSO2 product containers
+   (i.e. `/home/wso2carbon/wso2-artifact-volume`) for this purpose.
+   
+   In Carbon Kernel version 4 based products, the default `client-truststore.jks` truststore is located at the
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]: Same as before for this and the other point

File: deploy/Managing_Keystores_And_Truststores.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+   `<WSO2_PRODUCT_HOME>/repository/resources/security` directory. Thus, you may mount the volume containing the
+   custom keystore to `/home/wso2carbon/wso2-artifact-volume/repository/resources/security` directory.
+   
+   In Carbon Kernel version 5 based products, the default `client-truststore.jks` truststore is located at the
+   `<WSO2_PRODUCT_HOME>/resources/security` directory. Thus, you may mount the volume containing the custom keystore
+   to `/home/wso2carbon/wso2-artifact-volume/resources/security` directory.
+    
+   > In Kubernetes based WSO2 product deployments, we recommend you to use a ConfigMap resource to add the custom keystore
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]: product deployment, it is advisable that you used
```suggestion
   > In Kubernetes based WSO2 product deployment, it is advisable that you use a ConfigMap resource to add the custom keystore
```

File: deploy/Managing_Keystores_And_Truststores.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+   > and truststore.
+   > 
+   > * [Create a ConfigMap resource from the keystore file](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-files)
```
Comment (chamindi-a) [inline deploy/Managing_Keystores_And_Truststores.md]:
```suggestion
   > * For further understanding peruse:
   

- [Create a ConfigMap resource from the keystore file](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/#create-configmaps-from-files)

```

**Review by chamindi-a — CHANGES_REQUESTED** [review]
@chirangaalwis: made some suggestion, let me know if you need more clarifications

File: deploy/Managing_Keystores_And_Truststores.md [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+   
+   In Carbon Kernel **version 4** based products, the default `client-truststore.jks` truststore is located at the
+   `<WSO2_PRODUCT_HOME>/repository/resources/security` directory. Thus, you may mount the volume containing the
+   custom keystore to `/home/wso2carbon/wso2-artifact-volume/repository/resources/security` directory.
+   
+   In Carbon Kernel **version 5** based products, the default `client-truststore.jks` truststore is located at the
+   `<WSO2_PRODUCT_HOME>/resources/security` directory. Thus, you may mount the volume containing the custom keystore
+   to `/home/wso2carbon/wso2-artifact-volume/resources/security` directory.
+    
+   > In Kubernetes based WSO2 product deployment, it is advisable that you use a ConfigMap resource to add the custom keystore
```
Comment (ThilinaManamgoda) [inline deploy/Managing_Keystores_And_Truststores.md:66]: Would it be better if we mention the Kubernetes versions that support this approach? AFAIR it is from `1.10.0` right?.

Comment (chirangaalwis) [reply deploy/Managing_Keystores_And_Truststores.md:66]: @ThilinaManamgoda +1. Will add.

Comment (chirangaalwis) [reply deploy/Managing_Keystores_And_Truststores.md:66]: Added in https://github.com/wso2/container-guide/pull/15/files#diff-9bf3730dcb649a313efa880c13afc382R69.

**Review by chamindi-a — APPROVED** [review]

**Review by ThilinaManamgoda — APPROVED** [review]

**Pull request merged**